pkg/utils/sync_map.go

```diff
@@ -0,0 +1,87 @@
+/*
```

Comment:
looks good to me. Seems the new wrapper adds typesafe, length tracking and some helpers like `keys()` etc. I suggest to check in those changes earlier in a separate PR next time. which makes this PR easier to review.